src/geode/mesh/core/geode_regular_grid_solid.cpp

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2019 - 2022 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/mesh/core/geode_regular_grid_solid.h>
+
+#include <fstream>
+
+#include <geode/basic/bitsery_archive.h>
+#include <geode/basic/logger.h>
+#include <geode/basic/pimpl_impl.h>
+
+#include <geode/geometry/point.h>
+
+#include <geode/mesh/builder/regular_grid_solid_builder.h>
+#include <geode/mesh/core/private/grid_impl.h>
+#include <geode/mesh/core/private/points_impl.h>
+#include <geode/mesh/core/regular_grid_solid.h>
+
+namespace
+{
+    static constexpr std::array< std::array< geode::local_index_t, 3 >, 8 >
+        cell_vertices_translations{ { { 0, 0, 0 }, { 1, 0, 0 }, { 0, 1, 0 },

```

Comment:
translations? what does it mean?

Reply:
When you have a CellIndices, it is how much you need to increment your indices to have the VertexIndices.